integration_tests/suite/helpers/base.py

```diff
@@ -12,48 +11,58 @@
 from .database import DbHelper
 from .filesystem import FileSystemClient
 from wazo_test_helpers.asset_launching_test_case import (
-    AssetLaunchingTestCase,
+    AbstractAssetLaunchingHelper,
+    cached_class_property,
     NoSuchPort,
     NoSuchService,
     WrongClient,
 )
 from wazo_test_helpers import until
 
-use_asset = pytest.mark.usefixtures
 
-
-class BaseAssetLaunchingTestCase(AssetLaunchingTestCase):
-
-    assets_root = os.path.join(os.path.dirname(__file__), '../..', 'assets')
+class BaseAssetLaunchingHelper(AbstractAssetLaunchingHelper):
+    assets_root = Path(__file__).parent / '..' / '..' / 'assets'
     asset = 'base'
     service = 'agid'
 
     @classmethod
-    def make_agid(cls):
+    def reset_clients(cls):
+        for attr in ('agid', 'db', 'calld', 'confd', 'agentd', 'filesystem'):
+            delattr(cls, attr)
+        until.true(cls.agid.is_ready, timeout=30)
+
+    @classmethod
+    def launch_service_with_asset(cls):
+        """Make sure Agid service is up before starting first test."""
+        super().launch_service_with_asset()
+        until.true(cls.agid.is_ready, timeout=30)
+
+    @cached_class_property
+    def agid(cls) -> AgidClient:
         port = cls.service_port(4573, 'agid')
         return AgidClient('127.0.0.1', port)
 
-    @classmethod
-    def make_confd(cls):
+    @cached_class_property
+    def confd(cls) -> ConfdMockClient:
         port = cls.service_port('9486', 'confd')
         return ConfdMockClient('127.0.0.1', port, version='1.1')
 
-    @classmethod
-    def make_agentd(cls):
+    @cached_class_property
+    def agentd(cls) -> AgentdMockClient:
         return AgentdMockClient('127.0.0.1', cls.service_port('9493', 'agentd'))
 
-    @classmethod
-    def make_calld(cls):
+    @cached_class_property
+    def calld(cls) -> CalldMockClient:
         return CalldMockClient('127.0.0.1', cls.service_port('9500', 'calld'))
 
-    @classmethod
-    def make_database(cls):
+    @cached_class_property
+    def db(cls) -> DbHelper | WrongClient:
         try:
             port = cls.service_port(5432, 'postgres')
         except (NoSuchService, NoSuchPort):
             return WrongClient('postgres')
```

Comment:
Does the `WrongClient` still make sense?
From what I understand, the `cached_class_property` and the `WrongClient` play the same role, i.e. make the client lazy and fail only on the first usage

Reply:
Oh OK, I wasn't sure the overlap was exactly the same, but no, it doesn't really make sense unless it is used elsewhere and it's too complicated to change.

Reply:
Not used elsewhere, you can safely remove it :crossed_fingers: